src/mlstacks/utils/yaml_utils.py

```diff
@@ -0,0 +1,106 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for loading YAML files into Python objects."""
+from pathlib import Path
+from typing import Any, Dict, Union, cast
+
+import yaml
+
+from mlstacks.models.component import (
+    Component,
+    ComponentMetadata,
+)
+from mlstacks.models.stack import Stack
+
+
+def load_yaml_as_dict(path: Union[Path, str]) -> Dict[str, Any]:
+    """Loads a yaml file as a dictionary.
+
+    Args:
+        path: The path to the yaml file.
+
+    Returns:
+        The dictionary representation of the yaml file.
+    """
+    if type(path) == Path:
+        path = str(path)
+
+    if not Path(path).exists():
+        raise FileNotFoundError(f"File {path} not found.")
+
+    with open(path) as yaml_file:
+        yaml_dict = yaml.safe_load(yaml_file)
+
+    if yaml_dict is None:
+        yaml_dict = {}
+    return cast(Dict[str, Any], yaml_dict)
```

Comment:
this might not be a dict actually, could just be a single number for instance, that's why mypy is complaining. Better to handle this in code than to `cast`-silence mypy